eng/packages/General.props

```diff
@@ -5,10 +5,10 @@
     <PackageVersion Include="Microsoft.Bcl.HashCode" Version="1.1.1" />
     <PackageVersion Include="Microsoft.Bcl.TimeProvider" Version="$(MicrosoftBclTimeProviderVersion)" />
     <PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.Common" Version="4.4.0" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.4.0" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.4.0" />
-    <PackageVersion Include="Microsoft.CodeAnalysis" Version="4.4.0" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.Common" Version="$(MicrosoftCodeAnalysisVersion_4_4)" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion_4_4)" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="$(MicrosoftCodeAnalysisVersion_4_4)" />
+    <PackageVersion Include="Microsoft.CodeAnalysis" Version="$(MicrosoftCodeAnalysisVersion_4_4)" />
```

Comment:
In [dotnet/runtime](https://github.com/search?q=repo%3Adotnet%2Fruntime%20MicrosoftCodeAnalysisVersion_LatestVS&type=code) all packages that affect the VS design-time experience use `MicrosoftCodeAnalysisVersion_LatestVS` property
![image](https://github.com/user-attachments/assets/d3d44275-a028-462b-a61b-efef66cba819)

There's also a comment stating the reason: https://github.com/dotnet/runtime/blob/59c2ea578bd615a63d56e8ff4b1de0a6b824691f/eng/Versions.props#L40-L44

Reply:
This is to keep this PR focused only on Logging. If we bump to latest, then there's other areas which would needing updated to resolve warnings. We can do another PR which addresses all the warnings in other non-logging related generators (options, meter, etc).

Reply:
Sounds good.